_posts/2020-01-30-new-age-of-bastion.md

```diff
@@ -0,0 +1,165 @@
+```rust
+// Here we are specifying the used restart strategy for our supervisor.
+// By default the bastion's supervisors are always trying to restart
+// failed actors with unlimited amount of tries.
+//
+// At the beginning we're creating a new instance of RestartStrategy
+// and then provides a policy and a back-off strategy.
+let restart_strategy = RestartStrategy::default()
+    // Set the limits for supervisor, so that it could stop
+    // after 3 attempts. If the actor can't be started, the supervisor
+    // will remove the failed actor from tracking.
+    .with_restart_policy(RestartPolicy::Tries(3))
+    // Set the desired restart strategy. By default supervisor will
+    // try to restore the failed actor as soon as possible. However,
+    // in our case we want to restart with a small delay between the
+    // tries. Let's say that we want a regular time interval between the
+    // attempts which is equal to 1 second.
+    .with_actor_restart_strategy(ActorRestartStrategy::LinearBackOff {
+      timeout: Duration::from_secs(1),
+    });
+
+// Define the supervisor
+supervisor
+	 // That uses our restart strategy defined earlier
+	 .with_restart_strategy(restart_strategy)
+	 // And tracks the child group, defined in the following function
+	 .children(|children| failed_actors_group(children))
+```
+1- Now we have a signature method to address all the actors in the system. Check our context addressing system [here](https://docs.rs/bastion/0.3.4/bastion/context/struct.BastionContext.html#method.signature)
+
+```rust
+Bastion::children(|children| {
+    children.with_exec(|ctx: BastionContext| {
+        async move {
+            ctx.tell(&ctx.signature(), "Hello to myself");
+
+            Ok(())
+        }
+    })
+}).expect("Couldn't create the children group.");
+```
+
+2- We've added meta-architecture instantiation. What is that? So now you can have a boilerplate free hierarchy for your actors and/or workers.
+```rust
+let children = children! {
+	// the default redundancy is 1
+	redundancy: 100,
+	action: |msg| {
+	   // do something with the message here
+	},
+};
+
+// simpler supervisor declaration.
+let sp = supervisor! {};
+
+let children = children! {
+  	supervisor: sp,
+  	redundancy: 10,
+  	action: |msg| {
+    	// do something with the message here
+    },
+};
+```
+
+3- You can mark your blocking code and execute that code inside blocking adaptive thread pool of Bastion.
+```rust
+let task = blocking! {
+	thread::sleep(time::Duration::from_millis(3000));
+};
+```
+
+4- What other frameworks and implementation calls `block_on` we call it `run`.  Because that's how other ecosystems call it. We believe general conventions should be over the specific conventions. If you want to run a process and wait for it's completion. (e.g. the blocking task written above) use:
```

Comment:
If you want to run a process and wait for it's completion, you can call `run!`. `run` is equivalent to what other Rust frameworks call `block_on`. We decided to call it run because that's the name implementations in other languages use. We think sticking to cross languages conventions will avoid confusion among users, regardless of their favorite language.

With the blocking task written above, we can use: